src/app/teacher/create-tag-dialog/create-tag-dialog.component.ts

```diff
@@ -45,14 +62,31 @@ export class CreateTagDialogComponent implements OnInit {
     this.subscriptions.unsubscribe();
   }
 
+  private createUniqueTagValidator(): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+      return this.tags != null &&
+        this.projectTagService.doesTagAlreadyExist(this.tags, control.value)
```

Comment:
Do we need this function to be defined in ProjectTagService? Since it's only used in this component, we can just define it in this same class?
```suggestion
        this.doesTagAlreadyExist(control.value)
```


Private function in CreateTagDialogComponent (untested):
```
  private doesTagAlreadyExist(tagText: string): boolean {
    return this.tags.some((tag: Tag) => tag.text.toLowerCase().trim() === tagText.toLowerCase().trim());
  }
```

Also, do we need to trim the existing tags? Would they already be trimmed?